### PR TITLE
mcs: init cluster for scheduler

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

We need a cluster to init scheduler controller, and the cluster should implement the shared cluster interfaces.

```commit-message
```

### Check List

Tests

- Manual test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]

File: pkg/mcs/scheduling/server/config.go [hunk 1]
```diff
@@ -170,3 +196,210 @@ func (c *Config) Validate() error {
...
+// IsPlacementRulesCacheEnabled returns if the placement rules cache is enabled.
+func (o *PersistConfig) IsPlacementRulesCacheEnabled() bool {
+	return false
+}
+
+// SetPlacementRulesCacheEnabled sets if the placement rules cache is enabled.
+func (o *PersistConfig) SetPlacementRulesCacheEnabled(b bool) {}
+
+// SetEnableWitness sets if the witness is enabled.
+func (o *PersistConfig) SetEnableWitness(b bool) {}
```
Comment (lhy1024) [inline pkg/mcs/scheduling/server/config.go:405]: Empty function?

File: pkg/mcs/scheduling/server/server.go [hunk 2]
```diff
@@ -410,10 +510,62 @@ func (s *Server) startServer() (err error) {
 		log.Error("failed to register the service", zap.String("service-name", utils.SchedulingServiceName), errs.ZapError(err))
 		return err
 	}
+
 	atomic.StoreInt64(&s.isRunning, 1)
 	return nil
 }
 
+func (s *Server) waitAPIServiceReady() error {
```
Comment (lhy1024) [inline pkg/mcs/scheduling/server/server.go:518]: How about moving waitAPIServiceReady and isAPIServiceReady to util.go?

File: pkg/mcs/scheduling/server/config.go (hunk 1 above)
Comment (rleungx) [reply pkg/mcs/scheduling/server/config.go]: Yes, since we don't need them.

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**